WooCommerce/src/androidTest/kotlin/com/woocommerce/android/e2e/tests/ui/ProductsRealAPI.kt

```diff
@@ -60,14 +62,53 @@ class RealAPISandbox : TestBase() {
 
     @After
     fun tearDown() {
+        ProductListScreen()
+            .leaveSearchMode()
+
         WelcomeScreen
             .logoutIfNeeded(composeTestRule)
     }
 
     @Test
```

Comment:
instead of creating a new file for tests that use mocked vs real API, how about we use annotation on the test to differentiate the two? that way we maintain only 1 test file which could run both, since the previous "to-use-or-not-to-use" mock implementation is available on a scenario level i think we could do that. 

i'm thinking since what we care about most is coverage (not if the tests are using mocks or not) having them in separate files for the same feature there is a possibility of having duplicated coverage (one using mocks and one using real API).

the repo is still on[ junit 4](https://junit.org/junit4/javadoc/4.12/org/junit/experimental/categories/Categories.html) and looks like we can do this by adding `@Category` to the test something like
```suggestion
    @Test @Category ({RealAPITests.class})
```

more of an aside, on[ junit 5](https://junit.org/junit5/docs/5.0.2/api/org/junit/jupiter/api/Tag.html) it seems a lot more straightforward so we should also probably look into updating to that soon if it's going to be helpful for tests

Disclaimer: recommendation not tested

Reply:
I've also thought about having them in one file. However, I can think of a case when at some point we will either drop WireMock, or will drop the real API tests, and it'll be much faster if tests are separated. I can also think of a case when we need to run only certain kind of tests on CI, so having them in separate files or even packages (at some point) will allow specifying what we need to run via `test-target` parameter.

Your point about accidentally having the same test for both mocked and real APIs remains true, though. With the current amount of tests (4) and projected ~25, I would say that this probability is quite low, if the person writing the test pays attention.

Please let me know if the first paragraph makes sense and if it answers the question, @jostnes 🙂

Reply:
the way i imagine it is that, when/if we decide to drop either wiremock or real API tests, the test shouldn't be dropped but should be adjusted to work against the other API response type. so that way we don't lose test coverage. 

but since we're still early days, i'm okay with keeping it separate too. i still think it would be good to explore the possibility of using junit's`@Tag` / `@Category` annotations as it would give us the flexibility to run scenarios from separate test files in one test run (alternative to `test-target` which targets the whole file).